src/core/net/netif.hpp

```diff
@@ -156,6 +157,41 @@ class NetifMulticastAddress : public otNetifMulticastAddress,
     bool Matches(const Address &aAddress) const { return GetAddress() == aAddress; }
 };
 
+class ExternalNetifMulticastAddress : public NetifMulticastAddress
+{
+    friend class Netif;
+    friend class LinkedList<ExternalNetifMulticastAddress>;
+
+public:
+#if (OPENTHREAD_FTD || OPENTHREAD_MTD) && OPENTHREAD_CONFIG_MLR_ENABLE
+    /**
+     * This method returns the current Multicast Listener Registration (MLR) state.
+     *
+     * @returns The current Multicast Listener Registration state.
+     *
+     */
+    MlrState GetMlrState(void) const { return mMlrState; }
+
+    /**
+     * This method sets the Multicast Listener Registration (MLR) state.
+     *
+     * @param[in] aState  The new Multicast Listener Registration state.
+     *
+     */
+    void SetMlrState(MlrState aState) { mMlrState = aState; }
+#endif
+
+private:
+    ExternalNetifMulticastAddress *GetNext(void)
+    {
+        return static_cast<ExternalNetifMulticastAddress *>(const_cast<otNetifMulticastAddress *>(mNext));
+    }
+
+#if (OPENTHREAD_FTD || OPENTHREAD_MTD) && OPENTHREAD_CONFIG_MLR_ENABLE
```

Comment:
is `(OPENTHREAD_FTD || OPENTHREAD_MTD)` necessary?

Reply:
removed